ccm_web/client/src/components/SectionSelectorWidget.tsx

```diff
@@ -391,101 +403,140 @@ function SectionSelectorWidget (props: ISectionSelectorWidgetProps): JSX.Element
-          </Grid>
+          {
+            (props.search.length > 0 && searcher?.isInteractive) && (
+              <Grid item container className={classes.searchContainer} xs={12}>
+                <TextField
+                  className={classes.searchTextField}
+                  disabled={isSearching || isIniting}
+                  onChange={searchChange}
+                  value={searchFieldText}
+                  id='textField_Search'
+                  size='small'
+                  label={searchFieldLabel}
+                  variant='outlined'
+                  inputProps={{ maxLength: 256 }}
+                  InputProps={{ endAdornment: getSearchTextFieldEndAdornment(searchFieldText.length > 0) }}
+                />
+              </Grid>
+            )
+          }
           <Grid item container style={{ paddingLeft: '16px' }}>
-            <Grid item {...gridSpacing.title} className={classes.title}>
-              <Typography variant='h6' style={{ visibility: props.header?.title !== undefined ? 'visible' : 'hidden' }}>{props.header?.title}
-                <span hidden={props.selectedSections.length === 0}>
-                  ({props.selectedSections.length})
-                </span>
-              </Typography>
-            </Grid>
-            <Grid item {...gridSpacing['select all']}>
-              <FormGroup row style={checkboxStyle()}>
-                <FormControlLabel
-                control={
-                    <Checkbox
-                      checked={isSelectAllChecked}
-                      onChange={handleSelectAllClicked}
-                      name="selectAllUnstagedCB"
-                      color="primary"
+            {
+              props.header?.title !== undefined && (
+                <Grid item {...gridSpacing.title} className={classes.title}>
+                  <Typography variant='h6'>
+                    {props.header.title}
+                    {props.selectedSections.length > 0 && (
+                      <span> ({props.selectedSections.length})</span>
+                    )}
+                  </Typography>
+                </Grid>
+              )
+            }
+            {
+              props.multiSelect && (
+                <Grid item {...gridSpacing['select all']}>
+                  <FormGroup row style={checkboxStyle()}>
+                    <FormControlLabel
+                      control={
+                        <Checkbox
+                          checked={isSelectAllChecked}
+                          onChange={handleSelectAllClicked}
+                          name='selectAllUnstagedCB'
+                          color='primary'
+                        />
+                      }
+                      disabled={selectableSections().length === 0}
+                      label='Select All'
```

Comment:
See item 4 in PR description (conditional rendering).

Reply:
```
 // TODO display:none, visibility:hidden.. find a better way
```
@chrisrrowland, tagging you since I thought you might be somewhat interested in what I did here (as well as in the accessibility changes).

Reply:
@ssciolla, what do you mean by TODO here?

Reply:
Chris had made this comment, which I'm now removing because the issue is resolved. Item 4 explains the topic. Basically Chris had been using `display: none` and `visilibity: hidden` CSS to hide parts of the component that were only sometimes used. Instead, I'm just not rendering those elements at all if the settings aren't passed to the component.